homeassistant/helpers/device_registry.py

```diff
@@ -907,6 +907,13 @@ async def startup_clean(event: Event) -> None:
 
     hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, startup_clean)
 
+    @callback
+    def _on_homeassistant_stop(event: Event) -> None:
+        """Cancel debounced cleanup."""
+        debounced_cleanup.async_cancel()
```

Comment:
This sounds like something we should maybe add to the debouncer ?

Reply:
Adding it to the debouncer would be easy to implement, but I'm not sure that's the cleanest way...
IMO the cleanup should be done in the same place as where it was initialised.